hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/pipeline/WritableRatisContainerProvider.java

```diff
@@ -103,7 +110,13 @@ public ContainerInfo getContainer(final long size,
               excludeList);
         }
         if (containerInfo != null) {
-          return containerInfo;
+          // if containerID == -1, means Container allocation
+          // failed on selected pipeline
+          if (containerInfo.getContainerID() != -1) {
+            return containerInfo;
+          } else {
+            excludeList.addPipeline(containerInfo.getPipelineID());
```

Comment:
Trying to exclude the faulty pipeline may not work, because:

https://github.com/apache/ozone/blob/d1a92b94a807808f3c732cd55b9a0d5f25bdff20/hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/pipeline/WritableRatisContainerProvider.java#L192-L196